src/Core/SharedKernels/Enumerator/Enumerator/ViewModels/InterviewDetails/Questions/CategoricalComboboxAutocompleteViewModel.cs

```diff
@@ -158,29 +200,36 @@ public void ResetFilterAndOptions()
         {
             this.FilterText = null;
             
-            var suggestions = this.GetSuggestions(null).ToList();
+            //var suggestions = this.GetSuggestions(null);
             this.QuestionState.Validity.ExecutedWithoutExceptions();
 
             this.InvokeOnMainThread(() =>
             {
-                this.AutoCompleteSuggestions = suggestions;
+                //this.AutoCompleteSuggestions = suggestions;
+                this.AutoCompleteSuggestions = null;
+                //this.AutoCompleteSuggestions = null;
                 this.RaisePropertyChanged(nameof(this.FilterText));
             });
         }
 
-        private IEnumerable<OptionWithSearchTerm> GetSuggestions(string filter)
+        private List<OptionWithSearchTerm> GetSuggestions(string filter)
         {
             List<CategoricalOption> filteredOptions = this.filteredOptionsViewModel.GetOptions(filter, this.excludedOptions, 20);
 
-            foreach (var model in filteredOptions.Count == 1 && displaySelectedValue
+            var categoricalOptions = filteredOptions.Count == 1 && displaySelectedValue
                 ? filteredOptions
-                : filteredOptions.Where(x => !this.excludedOptions.Contains(x.Value)))
+                : filteredOptions.Where(x => !this.excludedOptions.Contains(x.Value));
+
+            var options = new List<OptionWithSearchTerm>();
+            foreach (var model in categoricalOptions)

```

Comment:
Could it be transformed to a shorter Linq expression?

Reply:
Done